subsys/net/ip/dhcpv4.c

```diff
@@ -745,6 +746,43 @@ static enum net_verdict parse_options(struct net_if *iface,
 			net_if_ipv4_set_gw(iface, &router);
 			break;
 		}
+		case DHCPV4_OPTIONS_DNS_SERVER: {
+			struct in_addr dns;
+			char dns_string[NET_IPV4_ADDR_LEN];
+			const char *dns_servers[] = {dns_string, NULL};
+
+			/* DNS server option may present 1 or more
+			* addresses. Each 4 bytes in length. DNS
+			* servers should be listed in order
+			* of preference.  Hence we choose the first
+			* and skip the rest.
+			*/
+			if (length % 4 != 0 || length < 4) {
+				NET_ERR("options_dns, bad length");
+				return NET_DROP;
+			}
+
+			frag = net_frag_read(frag, pos, &pos, 4, dns.s4_addr);
+			frag = net_frag_skip(frag, pos, &pos, length - 4);
+			if (!frag && pos) {
+				NET_ERR("options_dns, short packet");
+				return NET_DROP;
+			}
+
+			memset(dns_string, 0, sizeof(dns_string));
+			memcpy(dns_string, net_sprint_ipv4_addr(&dns),
```

Comment:
But this is what I don't like. Convert binary address to string, just for it to be converted back? Nope, we should allow to init resolver with struct in_addr.

Reply:
This definitely won't do, because net_sprint_ipv4_addr() is debugging func, available only when logging is enabled.

@pdxjohnny : I'm in the process of refactoring this patch already.